### DISCUSSION TITLE
How to wait with launching some program until another one has finished?

### DISCUSSION THREAD

**Discussion by mbork** [kovidgoyal/kitty#5685] [Q&A]
Hi all,

I want to write my session file so that it starts program A in tab 1 and then (after A finishes) starts program B in tab 2. How to do that?

(Rationale: A is starting a virtual machine and B is sshing into it, which won't work unless it is up.)

**Comment by page-down** [comment]
I would suggest starting a script directly in the second tab and repeatedly trying to connect to the VM via SSH.

The other ways are:
- [Allow remote control](https://sw.kovidgoyal.net/kitty/launch/#cmdoption-launch-allow-remote-control) in the first window where you start the VM, and create a second tab via the remote control command after the VM has been started.
- Use [watcher](https://sw.kovidgoyal.net/kitty/launch/#watchers) to handle the closing of the first window, check if the VM has started and create a second tab.
